PR: https://github.com/odoo/odoo/pull/114708

From: fceb3d8991d633e1004b69c5027c4846e933138e
From: Ahmed Khalaf
Date: 2023-06-01 09:43:43

Structural Changes: 6
Total Changes: 6

[IMP] mrp: enable precompute on MO field

This commit enables precompute of product_qty in MO so that the
product_qty is correctly set when a bom exists for the created MO.

it also corrects the values of `picking_type_use` fields to be used
in the barcode app for MOs.

Enterprise PR: odoo/enterprise#38096

Taskid: 3052744
Part-of: odoo/odoo#114708

================================= pseudo patch: =================================

--- a/addons/stock/models/stock_move_line.py
+++ b/addons/stock/models/stock_move_line.py
@@ -67,11 +67,11 @@ class StockMoveLine(models.Model):
     location_dest_usage = fields.Selection(string="Destination Location Type", related='location_dest_id.usage')
     lots_visible = fields.Boolean(compute='_compute_lots_visible')
     picking_partner_id = fields.Many2one(related='picking_id.partner_id', readonly=True)
-    picking_code = fields.Selection(related='picking_id.picking_type_id.code', readonly=True)
+    picking_code = fields.Selection(related='picking_type_id.code', readonly=True)
     picking_type_id = fields.Many2one(
         'stock.picking.type', 'Operation type', compute='_compute_picking_type_id', search='_search_picking_type_id')
-    picking_type_use_create_lots = fields.Boolean(related='picking_id.picking_type_id.use_create_lots', readonly=True)
-    picking_type_use_existing_lots = fields.Boolean(related='picking_id.picking_type_id.use_existing_lots', readonly=True)
+    picking_type_use_create_lots = fields.Boolean(related='picking_type_id.use_create_lots', readonly=True)
+    picking_type_use_existing_lots = fields.Boolean(related='picking_type_id.use_existing_lots', readonly=True)
     picking_type_entire_packs = fields.Boolean(related='picking_id.picking_type_id.show_entire_packs', readonly=True)
     state = fields.Selection(related='move_id.state', store=True, related_sudo=False)
     is_initial_demand_editable = fields.Boolean(related='move_id.is_initial_demand_editable')
